benchmark-service/benchmark/benchmark.py

```diff
@@ -215,27 +217,51 @@ async def compare(request, userdata):  # pylint: disable=unused-argument
     return await render_template('benchmark', request, userdata, 'compare.html', context)
 
 
-async def query_github(app):
-    global START_POINT
+async def update_commits(app):
+    global benchmark_data
     github_client = app['github_client']
-    request_string = f'/repos/hail-is/hail/commits?since={START_POINT}'
+    batch_client = app['batch_client']
+    gs_reader = app['gs_reader']
 
-    data = await github_client.getitem(request_string)
+    request_string = f'/repos/hail-is/hail/commits?since={START_POINT}'
+    gh_data = await github_client.getitem(request_string)
     new_commits = []
-    for commit in data:
-        sha = commit.get('sha')
-        new_commits.append(commit)
-        log.info(f'commit {sha}')
-        START_POINT = commit['commit']['author'].get('date')
-        log.info(f'start point is now {START_POINT}')
+    formatted_new_commits = []
+
+    for gh_commit in gh_data:
+        sha = gh_commit.get('sha')
+        file_path = f'gs://{HAIL_BENCHMARK_BUCKET_NAME}/benchmark-test/{sha}'
+        has_results_file = gs_reader.file_exists(file_path)
+
+        batches = [b async for b in batch_client.list_batches(q=f'sha={sha} running')]
+
+        if not batches and not has_results_file:
+            new_commits.append(gh_commit)
+            batch_id = await submit_test_batch(batch_client, sha)
+            batch = await batch_client.get_batch(batch_id)
+            batch_status = await batch.last_known_status()
+            log.info(f'submitted a batch {batch_id} for commit {sha}')
+            commit = {
```

Comment:
So something that is confusing to me is when benchmark restarts, won't you not have old commits in the `formatted_new_commits` and you're overwriting your old commits? Where is `new_commits` used? Maybe you want to break out this loop to something like this:

```
formatted_commits = []

for gh_commit in gh_data:
    sha = gh_commit.get('sha')
    file_path = f'gs://{HAIL_BENCHMARK_BUCKET_NAME}/benchmark-test/{sha}'
    has_results_file = gs_reader.file_exists(file_path)
    batches = [b async for b in batch_client.list_batches(q=f'sha={sha} running')]

    if not batches and not has_results_file:
         batch_id = await submit_test_batch(batch_client, sha)
         batch = await batch_client.get_batch(batch_id)
    else:
         batch = batches[:-1]

     commit = {

     }

     formatted_commits.append(commit)

benchmark_data = {
    'commits': formatted_commits
}